cdap-support-bundle/bin/collect_support_bundle.py

```diff
@@ -0,0 +1,655 @@
+# Copyright © 2023 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+# collect_support_bundle.py to get logs from CDAP instance deployed on kubernetes.
+
+import argparse
+import base64
+import json
+from json.decoder import JSONDecodeError
+import os
+import shutil
+import subprocess
+import sys
+from datetime import datetime
+
+parser = argparse.ArgumentParser(description='Log collector program for CDAP.')
+parser.add_argument(
+    "--namespace",
+    "-n",
+    help="Kubernetes namespace where CDAP is installed",
+    required=True,
+    type=str
+)
+parser.add_argument(
+    "--username",
+    "-u",
+    help="CDAP User Name",
+    type=str
+)
+parser.add_argument(
+    "--password",
+    "-p",
+    help="CDAP Password",
+    type=str
+)
+parser.add_argument(
+    "--cdap-ns",
+    "-c",
+    help="CDAP namespace",
+    type=str
+)
+parser.add_argument(
```

Comment:
If we're collecting logs for a pipeline run, both `pipeline-name` and `run-id` should be specified.
Otherwise (when we want to collect general CDAP debug data), neither `pipeline-name` nor `run-id` should be specified

Reply:
this is the current behaviour.